integration/experiment/ffnet/gen_neural_net.py

```diff
@@ -0,0 +1,266 @@
+                start_i = i*bs
+                end_i = start_i + bs
+                xb, yb = val_set[start_i:end_i]
+                total_loss += loss_fn(model(xb), yb[:,0]) * len(xb)
+        log(epoch, train_loss/n_samples, total_loss/n_test_samples)
+
+    return model
+
+def main(input_list, output_name="nnet", describe_net="A neural net.", region_ignore=None):
+    #Regions to ignore for training
+    if region_ignore == None:
+        region_list = []
+    else:
+        region_list = region_ignore.split(",")
+    region_ignore = ['NAN'] + region_list
+
+    dfs = []
+
+    region_ids = []
+
+    y_columns = ['region-id']
+    X_columns_domain = {
+            'cpu':['CPU_POWER-package-0',
+                'CPU_FREQUENCY_STATUS-package-0',
+                'MSR::UNCORE_PERF_STATUS:FREQ-package-0',
+                'MSR::QM_CTR_SCALED_RATE-package-0'],
+            'gpu':['GPU_CORE_FREQUENCY_STATUS-gpu-0',
+                'GPU_POWER-gpu-0',
+                'GPU_UTILIZATION-gpu-0',
+                'GPU_CORE_ACTIVITY-gpu-0',
+                'GPU_UNCORE_ACTIVITY-gpu-0']}
+    ratios_domain = {
+            'cpu': [['CPU_INSTRUCTIONS_RETIRED-package-0', 'TIME'],
+            ['CPU_CYCLES_THREAD-package-0', 'TIME'],
+            ['CPU_ENERGY-package-0', 'TIME'],
+            ['MSR::APERF:ACNT-package-0', 'MSR::MPERF:MCNT-package-0'],
+            ['MSR::PPERF:PCNT-package-0', 'MSR::MPERF:MCNT-package-0'],
+            ['MSR::PPERF:PCNT-package-0', 'MSR::APERF:ACNT-package-0'],
+            ],
+            'gpu':[]}
+
+    if type(input_list) == str:
+        input_list = [input_list]
+
+    for inp in input_list:
+        df = pd.read_hdf(inp)
+        if 'app-config' not in df:
+            sys.stderr.write('<geopm> Error: No app-config in input data. Have you used gen_hdf_from_fsweep.py to create this HDF?\n')
+            sys.exit(1)
+
+        df["region-id"] = df["app-config"]
+        df = df[~df['app-config'].isin(region_ignore)]
+        dfs.append(df)
+
+    df_traces = pd.concat(dfs)
+    domains_to_train = []
+
+# Check traces to determine which domains we can train on
+    for domain in X_columns_domain:
+        add_domain = True
+        for signal in set(X_columns_domain[domain] + sum(ratios_domain[domain], [])):
+            if signal not in df_traces.columns:
+                print (f"Missing signal {signal} is required for training {domain}.")
+                add_domain = False
+                break
+        if add_domain:
+            print (f"Training on {domain}")
+            domains_to_train.append(domain)
+
+    if len(domains_to_train) == 0:
+        sys.stderr.write('<geopm> Error: No training domain is complete\n')
+        sys.exit(1)
+
+    print("Training to identify these regions:")
+    region_ids = sorted(list(df_traces["region-id"].unique()))
+    print(", ".join(region_ids))
+    mapping = {region_id: region_ids.index(region_id) for region_id in region_ids}
```

Comment:
Can avoid running a search within each region_ids iteration:
```suggestion
    mapping = {region_id: idx for idx, region_id in enumerate(region_ids)}
```